### PR TITLE
Configure Duchy cluster.

### PR DESCRIPTION
**Pull request by wfa-hkawalkar**

### CODE REVIEW

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 5 unresolved discussions (waiting on @bdomen-ggl, @wfa-hkawalkar, and @wfa-siyengar)

___
*[`.github/workflows/configure-duchy.yml` line 41 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnbFYFBEsNQ5OvhnK5:-NRnbFYGB_5HYGoMHDma:btkpovw) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/c12abaa0983deeffca97db3372805808486c1713/.github/workflows/configure-duchy.yml#L41)):*
> ```YAML
> 
>       - name: Login to GitHub Packages
>         uses: docker/login-action@v1
> ```

This isn't necessary. The container images must already be pushed and accessible by the K8s node at this point.



_Code quote:_
```YAML
      - name: Login to GitHub Packages
        uses: docker/login-action@v1
        with:
          registry: ${{ env.REGISTRY }}
          username: ${{ github.actor }}
          password: ${{ secrets.GIT_SECRETS }}
```
___
*[`.github/workflows/configure-duchy.yml` line 55 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnblmv4NRSCUgK97pF:-NRnblmv4NRSCUgK97pG:bvhv94q) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/c12abaa0983deeffca97db3372805808486c1713/.github/workflows/configure-duchy.yml#L55)):*
> ```YAML
>       - name: Build K8s manifest
>         run: |
>           bazelisk build //src/main/k8s/dev:worker1_duchy_gke \
> ```

The Duchy name should be an input, as this workflow needs to be reusable for multiple Duchies.



_Code quote:_
```YAML
worker1
```
___
*[`.github/workflows/configure-duchy.yml` line 56 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRncfBz4MVnndtj-f3O:-NRncfBz4MVnndtj-f3P:blwetvt) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/c12abaa0983deeffca97db3372805808486c1713/.github/workflows/configure-duchy.yml#L56)):*
> ```YAML
>         run: |
>           bazelisk build //src/main/k8s/dev:worker1_duchy_gke \
>           --define k8s_duchy_secret_name=certs-and-configs \
> ```

This value is based on the result of the "Apply secrets" step. It'll be `certs-and-configs-<hash>` where the previous step generates a unique hash.



_Code quote:_
```YAML
certs-and-configs
```
___
*[`.github/workflows/configure-duchy.yml` line 57 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnc9TV5JRyglnXGrWc:-NRnc9TV5JRyglnXGrWd:b6dxmkb) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/c12abaa0983deeffca97db3372805808486c1713/.github/workflows/configure-duchy.yml#L57)):*
> ```YAML
>           bazelisk build //src/main/k8s/dev:worker1_duchy_gke \
>           --define k8s_duchy_secret_name=certs-and-configs \
>           --define duchy_cert_id=SVVse4xWHL0 \
> ```

This value will be different depending on the environment. For QA, it'll be the already existing certificate ID for this Duchy.



_Code quote:_
```YAML
SVVse4xWHL0
```
___
*[`.github/workflows/configure-duchy.yml` line 60 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnbf-V5UjSNBV2nMfE:-NRnbf-W-0cgpnen2a_Q:bsikry0) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/c12abaa0983deeffca97db3372805808486c1713/.github/workflows/configure-duchy.yml#L60)):*
> ```YAML
>           --define duchy_storage_bucket=worker1-duchy \
>           --define container_registry=${{ env.REGISTRY }} \
>           --define image_repo_prefix=world-federation-of-advertisers/packages \
> ```

nit: If REGISTRY is an environment variable, this should be too.



_Code quote:_
```YAML
world-federation-of-advertisers/package
```

**Review by wfa-siyengar — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 4 unresolved discussions (waiting on @bdomen-ggl, @SanjayVas, @wfa-hkawalkar, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 41 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnbFYFBEsNQ5OvhnK5:-NRxH4zC2uHrgp-sSnNl:bvpeo2a) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L41)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

This isn't necessary. The container images must already be pushed and accessible by the K8s node at this point.
</blockquote></details>

Removed. Done!
___
*[`.github/workflows/configure-duchy.yml` line 55 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnblmv4NRSCUgK97pF:-NRxJZB72GsFrwfwJ1VH:b-7cwawu) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L55)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

The Duchy name should be an input, as this workflow needs to be reusable for multiple Duchies.
</blockquote></details>

Done. It will be connected better when the caller workflow is in line.
___
*[`.github/workflows/configure-duchy.yml` line 56 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRncfBz4MVnndtj-f3O:-NRxLcr0DA_4kw6PUgz6:b-h5vu3s) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L56)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

This value is based on the result of the "Apply secrets" step. It'll be `certs-and-configs-<hash>` where the previous step generates a unique hash.
</blockquote></details>

@wfa-mdharmalingam @wfa-hkawalkar please check this and close.
___
*[`.github/workflows/configure-duchy.yml` line 57 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnc9TV5JRyglnXGrWc:-NRxK19FDwzURvNFb_g6:bagh4eh) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L57)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

This value will be different depending on the environment. For QA, it'll be the already existing certificate ID for this Duchy.
</blockquote></details>

Do you mean we hardcode it? or take this as input?
___
*[`.github/workflows/configure-duchy.yml` line 60 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnbf-V5UjSNBV2nMfE:-NRxLkT0DLrGYePeQE-o:b-6ozm1n) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L60)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

nit: If REGISTRY is an environment variable, this should be too.
</blockquote></details>

Done. Now repo prefix is an environment variable.

**Review by wfa-hkawalkar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 4 unresolved discussions (waiting on @bdomen-ggl, @SanjayVas, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 56 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRncfBz4MVnndtj-f3O:-NS-k6Pv4Mc4wJKwDyZ5:b6aret5) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L56)):*
<details><summary><i>Previously, wfa-siyengar wrote…</i></summary><blockquote>

@wfa-mdharmalingam @wfa-hkawalkar please check this and close.
</blockquote></details>

Yes, we will check on this.

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 4 unresolved discussions (waiting on @bdomen-ggl, @wfa-hkawalkar, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 57 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnc9TV5JRyglnXGrWc:-NS859Zn9zF8ebCWpND2:bee7n5j) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L57)):*
<details><summary><i>Previously, wfa-siyengar wrote…</i></summary><blockquote>

Do you mean we hardcode it? or take this as input?
</blockquote></details>

Take it as input at this level. Hard-code it at the higher-level QA workflow.

Basically the QA environment is expected to already exist with API resources. Therefore, we should already know the certificate IDs to use. For nightly, we'll need to read this value from the output of the resource setup job.
___
*[`.github/workflows/configure-duchy.yml` line 44 at r8](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NS84fjA3yAT5lEKOa0K:-NS84fjA3yAT5lEKOa0L:b-ycc1v2) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/78b407070da28546103c4b9f21d656049eefee77/.github/workflows/configure-duchy.yml#L44)):*
> ```YAML
>         uses: actions/checkout@v3
>         
>       # TODO(wfa-hkawalkar) add gcloud authentication.        
> ```

Resolve this comment when this is actually working.
___
*[`.github/workflows/configure-duchy.yml` line 49 at r8](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NS84kRE14RD5C9PUJBM:-NS84kRE14RD5C9PUJBN:beafw3f) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/78b407070da28546103c4b9f21d656049eefee77/.github/workflows/configure-duchy.yml#L49)):*
> ```YAML
>         run: bazelisk build //src/main/k8s/testing/secretfiles:apply_kustomization
> 
>       - name: Get Hash Value
> ```

See related comment in #921: this isn't the correct way to determine the secret name.

**Review by wfa-hkawalkar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @bdomen-ggl, @SanjayVas, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 44 at r8](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NS84fjA3yAT5lEKOa0K:-NS9sRlO5ySOvVJ3Ztkx:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/86a30fb3310220dd7fafe801c5834e4d43eff827/.github/workflows/configure-duchy.yml#L44)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Resolve this comment when this is actually working.
</blockquote></details>

Done.

**Review by wfa-hkawalkar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @bdomen-ggl, @SanjayVas, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 49 at r8](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NS84kRE14RD5C9PUJBM:-NSGLLbB2mIAxjtjeUyB:b-sgjolz) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/86a30fb3310220dd7fafe801c5834e4d43eff827/.github/workflows/configure-duchy.yml#L49)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

See related comment in #921: this isn't the correct way to determine the secret name.
</blockquote></details>

Changes made in the file as per PR#921.

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 3 unresolved discussions (waiting on @bdomen-ggl, @wfa-hkawalkar, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 19 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZBTw9Tj3Rm3cYRC9:-NSHZBTw9Tj3Rm3cYRCA:b-yc62o8) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/b0bea783bed18ecbf28f4b7ff7b709aa4c5de916/.github/workflows/configure-duchy.yml#L19)):*
> ```YAML
> on:
>   workflow_call:
>     workflow_dispatch:
> ```

`workflow_dispatch` is not a child of `workflow_call`.  I think something went wrong here in formatting.
___
*[`.github/workflows/configure-duchy.yml` line 43 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZJL00Jiu5w3VeDpX:-NSHZJL17QLFX_qi-25B:b8bqqdz) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/b0bea783bed18ecbf28f4b7ff7b709aa4c5de916/.github/workflows/configure-duchy.yml#L43)):*
> ```YAML
>       - name: Checkout code
>         uses: actions/checkout@v2
>         
> ```

nit: whitespace



_Code quote:_
```YAML
········
```

**Review by wfa-hkawalkar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 3 unresolved discussions (waiting on @bdomen-ggl, @SanjayVas, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 19 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZBTw9Tj3Rm3cYRC9:-NSHmAI75IA1oHksdtja:b3cxn54) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/588973d56a8a032757a3e9772cebd5d5709f7445/.github/workflows/configure-duchy.yml#L19)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

`workflow_dispatch` is not a child of `workflow_call`.  I think something went wrong here in formatting.
</blockquote></details>

Done. Thank you for pointing that out Sanjay.

**Review by wfa-mdharmalingam — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 2 unresolved discussions (waiting on @bdomen-ggl, @SanjayVas, and @wfa-hkawalkar)

___
*[`.github/workflows/configure-duchy.yml` line 43 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZJL00Jiu5w3VeDpX:-NSKBJU8BtD0HXfIU52_:bvm7z8d) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/8164c2f3e719bf984bf7e4be0d0e528da971ea86/.github/workflows/configure-duchy.yml#L43)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

nit: whitespace
</blockquote></details>

Resolved

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 2 unresolved discussions (waiting on @bdomen-ggl and @wfa-hkawalkar)

___
*[`.github/workflows/configure-duchy.yml` line 19 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZBTw9Tj3Rm3cYRC9:-NSMh1A69WHOsfjzqrqy:bxberiz) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/8164c2f3e719bf984bf7e4be0d0e528da971ea86/.github/workflows/configure-duchy.yml#L19)):*
<details><summary><i>Previously, wfa-hkawalkar wrote…</i></summary><blockquote>

Done. Thank you for pointing that out Sanjay.
</blockquote></details>

I don't see any change here.
___
*[`.github/workflows/configure-duchy.yml` line 43 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZJL00Jiu5w3VeDpX:-NSMh3blC8f6S5tlfimp:bxberiz) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/8164c2f3e719bf984bf7e4be0d0e528da971ea86/.github/workflows/configure-duchy.yml#L43)):*
<details><summary><i>Previously, wfa-mdharmalingam wrote…</i></summary><blockquote>

Resolved
</blockquote></details>

I don't see any change here.

**Review by wfa-mdharmalingam — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @bdomen-ggl, @SanjayVas, and @wfa-hkawalkar)

___
*[`.github/workflows/configure-duchy.yml` line 19 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZBTw9Tj3Rm3cYRC9:-NSNFm1j4dwY7LZARFt9:bnoc1l5) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/8164c2f3e719bf984bf7e4be0d0e528da971ea86/.github/workflows/configure-duchy.yml#L19)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

I don't see any change here.
</blockquote></details>

As confirmed, the workflow call will be put to use in a separate PR

**Review by SanjayVas — COMMENTED** [review]
> Reviewed 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 3 unresolved discussions (waiting on @bdomen-ggl and @wfa-hkawalkar)

___
*[`.github/workflows/configure-duchy.yml` line 43 at r11](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSO2lkT4dAMNMM5JTjP:-NSO2lkT4dAMNMM5JTjQ:b-pntkyh) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/e12a17d54e77240337318e6163d797d88ce265a3/.github/workflows/configure-duchy.yml#L43)):*
> ```YAML
>       - name: Checkout code
>         uses: actions/checkout@v2
>         
> ```

nit: trailing whitespace



_Code quote:_
```YAML
········
```
___
*[`.github/workflows/configure-duchy.yml` line 67 at r11](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSO2qFp-f2-_0153ynL:-NSO2qFp-f2-_0153ynM:b-pntkyh) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/e12a17d54e77240337318e6163d797d88ce265a3/.github/workflows/configure-duchy.yml#L67)):*
> ```YAML
>           resource_name="$(kubectl apply -k secretfiles/src/main/k8s/testing/secretfiles/kustomization -o name)"
>           echo "SECRET_NAME=$(basename $resource_name)" >> "$GITHUB_ENV"
>           
> ```

nit: trailing whitespace



_Code quote:_
```YAML
··········
```

**Review by wfa-hkawalkar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @bdomen-ggl and @SanjayVas)

___
*[`.github/workflows/configure-duchy.yml` line 43 at r10](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSHZJL00Jiu5w3VeDpX:-NSP-ix-4ROzmq8MIslF:b-w6rv78) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/8164c2f3e719bf984bf7e4be0d0e528da971ea86/.github/workflows/configure-duchy.yml#L43)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

I don't see any change here.
</blockquote></details>

Double checked it. removed white spaces now. 
___
*[`.github/workflows/configure-duchy.yml` line 43 at r11](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSO2lkT4dAMNMM5JTjP:-NSP00Xe6M6ZHobAe8eQ:b187adb) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/e12a17d54e77240337318e6163d797d88ce265a3/.github/workflows/configure-duchy.yml#L43)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

nit: trailing whitespace
</blockquote></details>

Whitespaces is removed now.

**Review by SanjayVas — COMMENTED** [review]
> Reviewed 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 1 unresolved discussion (waiting on @bdomen-ggl)

**Review by wfa-mdharmalingam — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 1 unresolved discussion (waiting on @bdomen-ggl and @SanjayVas)

___
*[`.github/workflows/configure-duchy.yml` line 57 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnc9TV5JRyglnXGrWc:-NSebaIg6jIaamGx710B:b-jqlyob) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L57)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Take it as input at this level. Hard-code it at the higher-level QA workflow.

Basically the QA environment is expected to already exist with API resources. Therefore, we should already know the certificate IDs to use. For nightly, we'll need to read this value from the output of the resource setup job.
</blockquote></details>

@wfa-siyengar Please look into this

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 1 unresolved discussion (waiting on @bdomen-ggl, @wfa-hkawalkar, and @wfa-mdharmalingam)

___
*[`.github/workflows/configure-duchy.yml` line 57 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnc9TV5JRyglnXGrWc:-NSg9fevBKBPJAlqsLZZ:b-b7le4q) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L57)):*
<details><summary><i>Previously, wfa-hkawalkar wrote…</i></summary><blockquote>

We will add this in the final PR.
</blockquote></details>

If that's the intent, then add a TODO here to take it as input.

To further explain:  A release triggers a top-level workflow specifying which environment to update. For QA, *that* workflow will have the specific certificate IDs for the QA environment hard-coded. *This* workflow should be environment-agnostic.

**Review by wfa-mdharmalingam — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @bdomen-ggl and @SanjayVas)

___
*[`.github/workflows/configure-duchy.yml` line 57 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRnc9TV5JRyglnXGrWc:-NSgRzSW8emj83CYcMAj:bc67b7o) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L57)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

If that's the intent, then add a TODO here to take it as input.

To further explain:  A release triggers a top-level workflow specifying which environment to update. For QA, *that* workflow will have the specific certificate IDs for the QA environment hard-coded. *This* workflow should be environment-agnostic.
</blockquote></details>

We have added the TODO to add Duchy as input on high level

**Review by wfa-siyengar — APPROVED** [review]
> Reviewed 1 of 1 files at r12, 1 of 1 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 1 unresolved discussion (waiting on @bdomen-ggl and @SanjayVas)

___
*[`.github/workflows/configure-duchy.yml` line 68 at r13](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NSgTyHg5myhGlfyNYB2:-NSiO8H_-jMJCB7igTsE:btr9mtd) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d72d0d741d191560ab7e5b5f9fc6024855805b4a/.github/workflows/configure-duchy.yml#L68)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

nit
</blockquote></details>

Although nit, i would like to understand this. 
Sorry, its not clear. Do you mean we should be taking Duchy certificate ID as an input from higher Level.

**Review by wfa-siyengar — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920)* status: all files reviewed, 1 unresolved discussion (waiting on @bdomen-ggl and @SanjayVas)

___
*[`.github/workflows/configure-duchy.yml` line 56 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/920#-NRncfBz4MVnndtj-f3O:-NSiP-E8EyIGfNs6p-XG:bz6own3) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/89a7420e27221eae17fbfc9997ccb7a623effea7/.github/workflows/configure-duchy.yml#L56)):*
<details><summary><i>Previously, wfa-hkawalkar wrote…</i></summary><blockquote>

Yes, we will check on this.
</blockquote></details>

@wfa-hkawalkar @wfa-mdharmalingam whats up with this one.
